webpack/JobWizard/steps/HostsAndInputs/index.js

```diff
@@ -0,0 +1,222 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Title,
+  Button,
+  Form,
+  FormGroup,
+  InputGroup,
+  Text,
+  Spinner,
+} from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import { FilterIcon } from '@patternfly/react-icons';
+import { debounce } from 'lodash';
+import { get } from 'foremanReact/redux/API';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { resetData } from 'foremanReact/components/AutoComplete/AutoCompleteActions';
+import {
+  selectJobTemplate,
+  selectWithKatello,
+  selectHostCount,
+} from '../../JobWizardSelectors';
+import { SelectField } from '../form/SelectField';
+import { SelectedChips } from './SelectedChips';
+import { TemplateInputs } from './TemplateInputs';
+import { HostSelect } from './HostSelect';
+import { HostSearch } from './HostSearch';
+import { HostPreviewModal } from './HostPreviewModal';
+import {
+  HOSTS,
+  HOST_COLLECTIONS,
+  HOST_GROUPS,
+  hostMethods,
+  hostsController,
+  hostQuerySearchID,
+  HOSTS_API,
+  hostsToPreviewAmount,
+} from '../../JobWizardConstants';
+
+const buildQuery = (selected, search) => {
+  const { hosts, hostCollections, hostGroups } = selected;
+  const hostsSearch = `(name = ${hosts.join(' or name = ')})`;
+  const hostCollectionsSearch = `(host_collection = ${hostCollections.join(
+    ' or host_collection = '
+  )})`;
+  const hostGroupsSearch = `(host_group = ${hostGroups.join(
+    ' or host_group = '
+  )})`;
+  return [
+    hosts.length ? hostsSearch : false,
+    hostCollections.length ? hostCollectionsSearch : false,
+    hostGroups.length ? hostGroupsSearch : false,
+    search.length ? `(${search})` : false,
+  ]
+    .filter(Boolean)
+    .join(' or ');
+};
+
+const HostsAndInputs = ({
+  templateValues,
+  setTemplateValues,
+  selected,
+  setSelected,
+  hostsSearchQuery,
+  setHostsSearchQuery,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  useEffect(() => {
+    debounce(() => {
+      setIsLoading(true);
+      dispatch(
+        get({
+          key: HOSTS_API,
+          url: '/api/hosts',
+          params: {
+            search: buildQuery(selected, hostsSearchQuery),
+            per_page: hostsToPreviewAmount,
+          },
+          handleSuccess: () => setIsLoading(false),
+        })
+      );
+    }, 700)();
```

Comment:
Might be better to make it larger to avoid multiple api calls

Reply:
can you also use a constant for it?